doc/specs/stdlib_hash_maps.md

```diff
@@ -0,0 +1,3112 @@
+`stdlib_hash_32bit`, and provides wrappers to some of the
+hash functions so that they no longer need to be supplied seeds. The
+module `stdlib_hashmap_chaining` defines a datatype,
+`chaining_hash_map_type`, implementing a simple separate chaining hash
+map noted more for its diagnostics than its performance. Finally the
+module, `stdlib_hashmap_open` defines a datatype,
+`open_hash_map_type`, implementing a simple open addressing hash
+map noted more for its diagnostics than its performance.
+
+These maps use separate chaining with linked lists and linear open
+addressing, respectively, to deal with hash index collisions, and are
+largely defined in the separated submodules, `stdlib_hashmap_chainings`
+and `stdlib_hashmap_opens`, respectively.
+In `chaining_hash_map_type` the colliding indices are handled by using
+linked lists with their roots at the hash index.
+In `open_hash_map_type`, the colliding indices are handled by searching
+from the initial hash index in increasing
+steps of one (modulo the hash map size) for an open map bin.
+
+The maps share many attributes in common. The two types share a
+common Application Programers Interface (API). The maps use powers of
+two for their slot sizes, so that the function, `fibonacci_hash`, can
+be used to map the hash codes to indices in the map. This is
+expected to be more efficient than prime number mapping using a
+modulo operation, and reduces the requirement that the hash
+function need to do a good job randomizing its lower order bits.
+This requires a good randomizing hash method for good performance.
+Both adjust the map size to reduce collisions, based on 
+the ratio of the number of hash map probes to the number of subroutine 
+calls.
+The maps make extensive use of pointers internally, but a private
+finalization subroutine avoids memory leaks.
+The maps can take entry keys of type `key_type`.
+Both maps allow the addition and lookup of entries, and the inclusion
+of data in addition to the entry key.
+The `chaining_hash_map_type` also allows the selective removal of
+entries.
+
+## The `stdlib_hashmap_wrappers` module
+
+The `stdlib_hashmap_wrappers` module provides data types to
+represent keys and associated data stored in a module, but is also, a
+wrapper for the `stdlib_hash_32bit` module. It allows
+direct access to the `stdlib_hash_32bit` procedures:
+`fibonacci_hash`, `fnv_1_hasher`, `fnv_1a_hasher`; and provides
+wrapper functions, `seeded_nmhash32_hasher`,
+`seeded_nmhash32x_hasher`, and `seeded_water_hasher` to the hashing
+functions: `nmhash32`, `nmhash32x`, and `water_hash`, respectively. It
+defines an interface, `hasher_fun`, compatible with the hash functions
+that take a `non-scalar key`. It defines one integer constant used
+as a kind value,`int_hash`. It also defines two types, `key_type` and
+`other_type`, and associated procedures, for storing and manipulating
+keys and their associated data.
+
+### The `stdlib_hashmap_wrappers` constant, `INT_HASH`
```

Comment:
```suggestion
### The `stdlib_hashmap_wrappers` constant, `int_hash`
```

Reply:
I have tried to change all upper case names to lower case. Let me know if I missed any.